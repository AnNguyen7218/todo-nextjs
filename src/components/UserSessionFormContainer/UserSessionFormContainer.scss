@import 'config.media';
@import 'config.gutters';

.container {
  display: flex;
  flex-direction: row;
  padding-top: gutter(48);
  padding-bottom: gutter(48);
  justify-content: space-around;

  @include break(lg) {
    padding-top: gutter(96);
    padding-bottom: gutter(192);
  }
}

.desktop-only {
  display: none;

  @include break(lg) {
    display: block;
  }
}

.img {
  width: 43rem;
}

.form-section {
  width: 100%;

  @include break(lg) {
    width: 34rem;
  }
}
