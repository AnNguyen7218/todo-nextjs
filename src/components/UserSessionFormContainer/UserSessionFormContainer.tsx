import React from 'react';
import cx from 'classnames';
import SessionFormImgSrc from 'assets/images/session-form.png';
import { Container } from 'components';
import css from './UserSessionFormContainer.scss';

type Props = {
  children: React.ReactChild;
  containerClassName?: string;
  formSectionClassName?: string;
};

const UserSessionFormContainer: React.FC<Props> = ({
  children,
  containerClassName,
  formSectionClassName,
}) => {
  return (
    <Container className={cx(css['container'], containerClassName)}>
      <div className={css['desktop-only']}>
        <img src={SessionFormImgSrc} className={css['img']} />
      </div>
      <div className={cx(css['form-section'], formSectionClassName)}>
        {children}
      </div>
    </Container>
  );
};

export default UserSessionFormContainer;
