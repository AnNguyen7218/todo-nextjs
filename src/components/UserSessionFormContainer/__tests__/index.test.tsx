import React from 'react';
import { render, cleanup } from '@testing-library/react';
import UserSessionFormContainer from '../UserSessionFormContainer';

afterEach(cleanup);

describe('UserSessionFormContainer Component', () => {
  test('it should match snapshot', () => {
    const { container } = render(
      <UserSessionFormContainer>
        <>
          <h2>Test form</h2>
          <input type="text" defaultValue="test value" />
        </>
      </UserSessionFormContainer>
    );
    expect(container).toMatchSnapshot();
  });
});
