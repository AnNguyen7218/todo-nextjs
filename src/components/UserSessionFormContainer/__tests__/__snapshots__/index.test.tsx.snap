// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSessionFormContainer Component it should match snapshot 1`] = `
<div>
  <div
    class="container container"
  >
    <div
      class="desktop-only"
    >
      <img
        class="img"
        src="session-form.png"
      />
    </div>
    <div
      class="form-section"
    >
      <h2>
        Test form
      </h2>
      <input
        type="text"
        value="test value"
      />
    </div>
  </div>
</div>
`;
